src/Core/Services/CipherService.cs

```diff
@@ -255,33 +255,28 @@ async Task<List<CipherView>> doTask()
                     {
                         throw new Exception("No key.");
                     }
-                    var decCiphers = new List<CipherView>();
-                    async Task decryptAndAddCipherAsync(Cipher cipher)
-                    {
-                        var c = await cipher.DecryptAsync();
-                        decCiphers.Add(c);
-                    }
-                    var tasks = new List<Task>();
-                    IEnumerable<Cipher> ciphers = await GetAllAsync();
+                    
+                    var parallelQuery = (await GetAllAsync()).AsParallel();
                     if (filter != null)
                     {
-                        ciphers = ciphers.Where(filter);
-                    }
-
-                    foreach (var cipher in ciphers)
-                    {
-                        tasks.Add(decryptAndAddCipherAsync(cipher));
+                        parallelQuery = parallelQuery.Where(filter);
                     }
-                    await Task.WhenAll(tasks);
-                    decCiphers = decCiphers.OrderBy(c => c, new CipherLocaleComparer(_i18nService)).ToList();
-
-                    if (filter != null)
+    
+                    var decCiphers = parallelQuery
+                        .Select(cipher =>  
+                        {
+                            return cipher.DecryptAsync().Result;
```

Comment:
Thanks for the article. I'm not too familiar with how c# handles async. I believe in this case it *might* not have been a problem, since every synchronous, blocking .Result is on a separate thread, but I might also be wrong about this.

Either way, I've changed it now to use async only.